### PR TITLE
Fontify yield and yield* as keyword

### LINKED ISSUE #67: font-lock yield keyword
`yield` and `yield*` are not highlighted as keywords. It's an easy workaround (below), but should probably be in this package

```elisp
(font-lock-add-keywords 'typescript-mode
                        '(("\\<\\(yield\\(\\*\\|\\>\\)\\)" . 'font-lock-keyword-face)))
```

### PR DESCRIPTION
**Pull request by felipeochoa**
Closes #67

### CODE REVIEW

File: typescript-mode.el [hunk 1]
```diff
@@ -318,6 +318,7 @@ Match group 1 is the name of the macro.")
                 (list "\\_<for\\_>"
                       "\\s-+\\(each\\)\\_>" nil nil
                       (list 1 'font-lock-keyword-face))
+                (cons "\\_<yield\\*?" 'font-lock-keyword-face)
                 (cons typescript--basic-type-re font-lock-type-face)
```
Comment (josteink) [inline typescript-mode.el]: This lacks a closing `\\_>`, or else "yield" in "**yield**er" too will be fontified as a keyword.

**Comment by josteink** [comment]
Hey there and thanks for the patch!

All in all it looks mostly good and I especially appreciate you adding a test-case for it too. That said, there's a small error there, which I've left a comment for.

If you could create a negative test-case for that one, and fix the regexp, I'd be more than happy to merge this!

**Comment by felipeochoa** [comment]
Ok updated!

**Comment by josteink** [comment]
Consider it merged! I simplified the regexp a little, but otherwise I kept things as is.

Thanks for the patch.

**Comment by josteink** [comment]
It looks like my simplification broke parts of your tests without me noticing before submitting: https://github.com/ananthakumaran/typescript.el/commit/526f307ca8ca50b8f98975f2812198212f6acf2f#diff-53b8a294602753ef50f7c7d7902490d5R362

I'm not quite sure I understand *how* that test should work, as such, I'm hesitant to rush a quick fix. Care to help me out? :)

**Pull request merged**